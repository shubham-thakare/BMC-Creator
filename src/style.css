@import "~@blueprintjs/core/lib/css/blueprint.css";
@import "~@blueprintjs/icons/lib/css/blueprint-icons.css";

:root {
  --app-background-color: #f5f8fa;
  --app-divider-color: #182026;
  --link-color: #182026;
  --link-hover-color: #818181;
  --white-color: #ffffff;
  --black-color: #000000;
  --red-color: #ff0000;
  --title-input-color: #000000;
  --title-input-placeholder-color: #8a9ba8;
  --placeholder-color: #8a9ba8;
  --title-input-font-size: 32px;
  --input-font-size: 14px;
  --navbar-link-font-size: 18px;
  --navbar-link-font-size-smaller: 14px;
  --input-font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "Icons16", sans-serif;
  --spacing-scale-1: 8px;
  --elevation-1: 0px 0px 5px -2px;
  --elevation-2: -1px 2px 5px 0px rgba(0, 0, 0, 0.2), 2px 2px 2px 0 rgba(0, 0, 0, 0.19);
}

* {
  box-sizing: border-box;
}

body {
  font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "Icons16", sans-serif;
  overflow-x: hidden;
  margin: 0;
  background: var(--app-background-color);
  user-select: none;
  -webkit-user-drag: none;
}

/* width */
::-webkit-scrollbar {
  width: 10px;
}

/* Track */
::-webkit-scrollbar-track {
  background: transparent;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: rgba(0, 0, 0, 0.25);
  -webkit-border-radius: 10px;
  border-radius: 10px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: rgba(0, 0, 0, 0.35);
}

span:focus {
  outline: none;
}

.bp3-editable-text.bp3-editable-text-placeholder:hover, .bp3-editable-text.bp3-editable-text-editing:hover {
  box-shadow: 0 0 0 0 rgba(19,124,189,0), 0 0 0 0 rgba(19,124,189,0), inset 0 0 0 1px rgba(16,22,26,0.15);
}

.bp3-editable-text.bp3-editable-text-placeholder:before, .bp3-editable-text.bp3-editable-text-editing:before {
  box-shadow: none;
}

.bp3-editable-text.bp3-editable-text-placeholder, .bp3-editable-text.bp3-editable-text-editing {
  padding: 5px;
  border-radius: 5px;
}

button.bp3-button {
  outline: none;
}

.full-width {
  width: 100%;
}

.text-bold {
  font-weight: bold;
}

.text-italic {
  font-style: italic;
}

.no-animation {
  animation: none !important;
}