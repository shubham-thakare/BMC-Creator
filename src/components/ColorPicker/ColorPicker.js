import React from 'react';
import PropTypes from 'prop-types';
import { ColorPickerWrapper } from './styles/ColorPicker.style';

const ColorPicker = ({ label, ...props }) => {
  return (
    <>
      <label for="colorwell">
        <b>{label}</b>
      </label>
      <ColorPickerWrapper {...props} list="true" name="colorwell" />
    </>
  );
};

ColorPicker.propTypes = {
  label: PropTypes.string,
};

export default ColorPicker;
