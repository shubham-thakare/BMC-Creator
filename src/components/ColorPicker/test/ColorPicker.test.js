import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from '..';
import { ColorPickerWrapper } from '../styles/ColorPicker.style';

let wrapper;

const props = {
  label: 'Testing Color Picker',
};

describe('ColorPicker Component', () => {
  beforeAll(() => {
    wrapper = shallow(<ColorPicker {...props} />);
  });

  it('To match snapshot of ColorPicker', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('Label exists', async (done) => {
    expect(wrapper.find('label')).toHaveLength(1);
    done();
  });

  it('ColorPickerWrapper exists', async (done) => {
    expect(wrapper.find(ColorPickerWrapper)).toHaveLength(1);
    done();
  });
});
