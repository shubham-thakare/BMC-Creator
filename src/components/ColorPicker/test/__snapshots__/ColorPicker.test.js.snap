// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Component To match snapshot of ColorPicker 1`] = `
<Fragment>
  <label
    htmlFor="colorwell"
  >
    <b>
      Testing Color Picker
    </b>
  </label>
  <ColorPickerstyle__ColorPickerWrapper
    list="true"
    name="colorwell"
  />
</Fragment>
`;
